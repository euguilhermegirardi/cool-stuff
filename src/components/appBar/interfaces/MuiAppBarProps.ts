export interface MuiAppBarProps {
  anchorElUser: HTMLElement | null;
  drawerWidth: number;
  mobileOpen: boolean;
  pages: string[];
  settings: string[];
  handleDrawerToggle: () => void;
  handleOpenUserMenu: (event: React.MouseEvent<HTMLElement>) => void;
  handleCloseUserMenu: () => void;
  handleCloseNavMenu: () => void;
};
