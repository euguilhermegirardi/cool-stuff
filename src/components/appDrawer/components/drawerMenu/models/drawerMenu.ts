import ApplicationRoutes from 'utils/navigation/applicationRoutes';
import translations from 'utils/translations';

export interface DrawerMenuProps {
  id: number;
  path: string;
  name: string;
};

export const drawerMenu: DrawerMenuProps[] = [
  {
    id: 1,
    path: ApplicationRoutes.dashboard,
    name: translations.dashboard.title
  },
  {
    id: 2,
    path: ApplicationRoutes.pokemonList,
    name: translations.pokemonList.title
  },
  {
    id: 3,
    path: ApplicationRoutes.pokemonCard,
    name: translations.pokemonCard.title
  },
  {
    id: 4,
    path: ApplicationRoutes.pokemonStatistics,
    name: translations.pokemonStatistics.title
  },
];
