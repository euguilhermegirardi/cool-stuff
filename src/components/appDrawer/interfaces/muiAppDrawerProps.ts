export interface MuiAppDrawerProps {
  drawerWidth: number;
  mobileOpen: boolean;
  handleDrawerToggle: () => void;
};
