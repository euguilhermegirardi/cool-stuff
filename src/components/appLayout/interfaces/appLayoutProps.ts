import { ReactElement } from 'react';

export interface AppLayoutProps {
  drawerWidth: number;
  mobileOpen: boolean;
  handleDrawerToggle: () => void;
  children: ReactElement;
};
