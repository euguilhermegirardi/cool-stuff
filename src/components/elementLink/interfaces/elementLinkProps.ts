export default interface ElementLinkProps {
  link: string;
  text: string;
  onClick?: () => void;
};
