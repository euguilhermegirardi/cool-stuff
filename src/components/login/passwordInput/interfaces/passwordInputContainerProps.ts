import { UseFormRegister } from 'react-hook-form';

export interface PasswordInputContainerProps {
  autoFocus: boolean;
  register: UseFormRegister<any>;
  formErrors?: {
    message?: string;
  };
  inputId: string;
  validateInput?: () => void;
};
