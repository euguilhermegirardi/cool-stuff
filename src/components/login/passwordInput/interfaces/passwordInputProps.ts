import { UseFormRegister } from 'react-hook-form';

export interface PasswordInputProps {
  autoFocus: boolean;
  register: UseFormRegister<any>;
  formErrors?: {
    message?: string;
  };
  showPassword: boolean;
  handleShowPassword: () => void;
  handleMouseDownPassword: (
    event: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => void;
  inputId: string;
};
