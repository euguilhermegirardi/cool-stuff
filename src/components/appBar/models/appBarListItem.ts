import { DrawerMenuProps } from 'components/appDrawer/components/drawerMenu/models/drawerMenu';
import ApplicationRoutes from 'utils/navigation/applicationRoutes';

export interface AppBarListItems extends DrawerMenuProps { };

export const appBarListItems: AppBarListItems[] = [
  {
    id: 1,
    path: ApplicationRoutes.projectTechnologies,
    name: 'Project Technologies',
  },
  {
    id: 2,
    path: ApplicationRoutes.contactUs,
    name: 'Contact Us',
  },
];
