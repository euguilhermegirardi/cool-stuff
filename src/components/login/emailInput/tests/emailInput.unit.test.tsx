import { useForm } from 'react-hook-form';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MockEmailInputProps } from './interfaces/emailInputProps';
import { mockEmptyFormErrors, mockFormErrors } from './models/emailInputProps';
import EmailInput from '../emailInput';

describe('Email Input Component', () => {
  const EmailInputTestContainer = ({
    formErrors,
  }: {
    formErrors: {
      email?: {
        message?: string;
      };
    }
  }) => {
    const { register } = useForm<MockEmailInputProps>({
      defaultValues: {
        test: 'joe doe',
      },
    });

    return (
      <EmailInput register={register} formErrors={formErrors} />
    );
  };

  const renderEmailInputComponent = (
    formErrors: {
      email?: {
        message?: string;
      };
    }
  ) =>
    render(<EmailInputTestContainer formErrors={formErrors} />);

  it('should render the component', () => {
    renderEmailInputComponent(mockEmptyFormErrors);
  });

  it('should render the input with the svg icon', () => {
    renderEmailInputComponent(mockEmptyFormErrors);
    const svgIcon = screen.getByTestId('MailOutlineIcon');
    const input = screen.getByRole('textbox');

    expect(svgIcon).toBeInTheDocument();
    expect(input).toBeInTheDocument();
  });

  it('should be able to type in the input', async () => {
    renderEmailInputComponent(mockEmptyFormErrors);
    const input = screen.getByRole('textbox');

    expect(input).toHaveValue('');
    await userEvent.type(input, 'testing the input');
    expect(input).toHaveValue('testing the input');
  });

  it('should render the form error message', () => {
    renderEmailInputComponent(mockFormErrors);
    const errorMessage = screen.getByText(mockFormErrors.email.message);

    expect(errorMessage).toBeInTheDocument();
  })
});
