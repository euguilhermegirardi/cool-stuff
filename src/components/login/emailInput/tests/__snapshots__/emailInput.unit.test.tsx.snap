// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Email Input Component > should match the snapshot with form errors 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
          style="height: 40px; border-radius: 0; margin-bottom: 8px;"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-vpsq89-MuiInputAdornment-root"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MailOutlineIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
              />
            </svg>
          </div>
          <input
            aria-invalid="true"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r4:"
            name="email"
            placeholder="email"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-180x7my-MuiTypography-root"
        role="alert"
      >
        email is required
      </p>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
        style="height: 40px; border-radius: 0; margin-bottom: 8px;"
      >
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-vpsq89-MuiInputAdornment-root"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MailOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
            />
          </svg>
        </div>
        <input
          aria-invalid="true"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r4:"
          name="email"
          placeholder="email"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-180x7my-MuiTypography-root"
      role="alert"
    >
      email is required
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Email Input Component > should match the snapshot without form errors 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
          style="height: 40px; border-radius: 0; margin-bottom: 8px;"
        >
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-vpsq89-MuiInputAdornment-root"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MailOutlineIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
              />
            </svg>
          </div>
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r5:"
            name="email"
            placeholder="email"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <p
        class="MuiTypography-root MuiTypography-body1 css-180x7my-MuiTypography-root"
        role="alert"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
        style="height: 40px; border-radius: 0; margin-bottom: 8px;"
      >
        <div
          class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-vpsq89-MuiInputAdornment-root"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MailOutlineIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 14H4V8l8 5 8-5v10zm-8-7L4 6h16l-8 5z"
            />
          </svg>
        </div>
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r5:"
          name="email"
          placeholder="email"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-180x7my-MuiTypography-root"
      role="alert"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
